### PR TITLE
Add Constant contact UI#157

### PR DESCRIPTION
**Pull request by karthicksakkaravarti**
New local.py settings:
CC_CLIENT_ID=""
CC_CLIENT_SECRET=""
CC_REDIRECT_URL = 'https://<host_name>/accounts/constantcontact/callback'

### CODE REVIEW

**Comment by karthicksakkaravarti** [comment]
Registering app on CC:

- Head over to https://app.constantcontact.com/pages/dma/portal/appList
- Click on the new application: 
Screenshot 2023-01-25 at 11 02 08 PM

- Click on the application which created newly
- Update Redirect URI : eg: https://dev2.weracehere.org/accounts/constantcontact/callback
- Click on generate client secret & copy the secret

Config:
CC_CLIENT_ID="<app_id>"
CC_CLIENT_SECRET="<client secret>"
CC_REDIRECT_URL = 'https://<host_name>/accounts/constantcontact/callback'

File: wrh_organization/apps/constant_contact/views.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            'grant_type': 'authorization_code'
+        },
+        headers={'Authorization': f"Basic {base64_string}"}
+        )
+        save_user_toek(request, res.json().get('access_token'))
+        return HttpResponseRedirect("/#/dashboard/member-profile")
+
+def save_user_toek(request, token):
+    current_user = request.user
+    current_user.more_data['cc_token'] = token
```
Comment (pprolancer) [inline wrh_organization/apps/constant_contact/views.py:33]: Hi @karthicksakkaravarti 
saving the token in a database might cause some issues.
the better way is to save the token in django session.
something like this:
```
request.session['cc_token'] = token
```

File: wrh_organization/apps/constant_contact/rest_api/views.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                return Response({"message": "Logged out successfully"}, status= status.HTTP_200_OK)
+        except Exception as e:
+            return Response({"message": "Something went wrong"}, status=status.HTTP_400_BAD_REQUEST)
+
+
+class CCStatusView(APIView):
+    permission_classes = []
+    def get(self, request , *args, **kwargs):
+        try:
+            if request.user.more_data and request.user.more_data.get('cc_token'):
```
Comment (pprolancer) [inline wrh_organization/apps/constant_contact/rest_api/views.py:28]: here also you can use that custom permission class

File: wrh_organization/apps/constant_contact/rest_api/views.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            else:
+                return Response({"is_singin": False}, status=status.HTTP_200_OK)
+        except Exception as e:
+            return Response({})
+
+class CCContactListView(APIView):
+    permission_classes = []
+    def get(self, request , *args, **kwargs):
+        try:
+            if request.user.more_data.get('cc_token', None):
```
Comment (pprolancer) [inline wrh_organization/apps/constant_contact/rest_api/views.py:40]: here also you can use that custom permission class

File: wrh_organization/apps/constant_contact/rest_api/views.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+            else:
+                return Response({'detail': 'Please authenticate with CC'}, status=status.HTTP_400_BAD_REQUEST)
+        except Exception as e:
+            return Response({})
+
+class CCContactListDetailView(APIView):
+    permission_classes = []
+    def get(self, request ,list_id,  *args, **kwargs):
+        url = f"https://api.cc.email/v3/contacts?lists={list_id}"
```
Comment (pprolancer) [inline wrh_organization/apps/constant_contact/rest_api/views.py:55]: here you forgot to check cc_token

File: wrh_organization/apps/constant_contact/rest_api/views.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+            else:
+                return Response({"is_singin": False}, status=status.HTTP_200_OK)
+        except Exception as e:
+            return Response({})
+
+class CCContactListView(APIView):
+    permission_classes = [ConstantContentLoginRequired]
+    def get(self, request , *args, **kwargs):
+        try:
+            if request.session.get('cc_token'):
```
Comment (pprolancer) [inline wrh_organization/apps/constant_contact/rest_api/views.py:37]: @karthicksakkaravarti 
as you have added `ConstantContentLoginRequired` permission class, then you dont need to check 'cc_token' in session anymore.

**Pull request merged**